lib/LedgerSMB/Scripts/payment.pm

```diff
@@ -163,42 +174,31 @@ sub pre_bulk_post_report {
         memo            => $request->{_locale}->text('Memo'),
         debits           => $request->{_locale}->text('Debits'),
         credits          => $request->{_locale}->text('Credits')
-                  };
-    my $total = 0;
-    for my $crow (1 .. $request->{contact_count}){
-        my $ref;
-        my $cid = $request->{"contact_$crow"};
-        if ($request->{"id_$cid"}){
-            $ref = {pay_to    => $request->{"contact_label_$cid"},
-                    accno     => $request->{ar_ap_accno},
-                    transdate => $request->{date_paid},
-                    source    => $request->{"source_$cid"},
-                    memo      => $request->{"memo_$cid"},
-                    amount    => 0
-                   };
-            for my $invrow (1 .. $request->{"invoice_count_$cid"}){
-                 my $inv_id = $request->{"invoice_${cid}_$invrow"};
-                 $ref->{amount} +=
-                    LedgerSMB::PGNumber->from_input($request->{"payment_$inv_id"});
-             }
-             # If vendor, this is debit-normal so multiply by -1
-             if ($request->{account_class} == EC_VENDOR ){ # vendor
-                 $ref->{amount} *= -1;
-              }
-              if ($ref->{amount} < 0) {
-                  $ref->{debits} = $ref->{amount} * -1;
-                  $ref->{credits} = 0;
-              } else {
-                  $ref->{debits} = 0;
-                  $ref->{credits} = $ref->{amount};
-              }
-              $total_debits += $ref->{debits};
-              $total_credits += $ref->{credits};
-              push @$rows, $ref;
-              $total += $ref->{amount};
-        }
+    };
+
+    # parse the flat "request" namespace into a hierarchical structure
+    # as defined by the $bulk_post_map transform
+    my $data = $bulk_post_map->($request);
+
+    # The user interface sets the 'id' field true-ish when the customer
+    # is selected for inclusion in the bulk payment
+    @{$data->{contacts}} = grep { $_->{id} } @{$data->{contacts}};
+    for my $crow (@{$data->{contacts}}) {
+        $crow->{accno} = $request->{ar_ap_accno};
```

Comment:
I think this could use `$data->{ar_ap_accno}` - would be more consistent with the approach of mapping all wanted request parameters into a hierarchical data structure. Same with the following line.